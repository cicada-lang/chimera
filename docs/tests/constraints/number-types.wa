print find _ {
  Number(1)
}

print find _ {
  Number("A")
}

print find x {
  x = "cat"
  Number(x)
}

print find x {
  x = 1
  Number(x)
}

print find x {
  Number(x)
  String(x)
}

// An inequality is subsumed by a type constraint.

print find x {
  x != "cat"
  Number(x)
}

// An inequality (applied later) is subsumed by a type constraint.

print find x {
  Number(x)
  x != "cat"
}

// An inequality (applied later) is subsumed by an indirect type constraint.

print find x {
  Number(z)
  z = x
  x != "cat"
}

print find x {
  Number(z)
  x != "cat"
  z = x
}

// An inequality (in a conjunction) is subsumed by a type constraint.

print find [x, y] {
  [x, y] != ["cat", 1]
  Number(x)
}

// An inequality (in a conjunction and applied later) is subsumed by a type constraint.

print find [x, y] {
  Number(x)
  [x, y] != ["cat", 1]
}
