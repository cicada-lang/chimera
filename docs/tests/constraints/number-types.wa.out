[_.0]
[]
[]
[1]
[]
[_.0]
[_.0]
[_.0]
[_.0]
[[_.0, _.1]]
[[_.0, _.1]]
